src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -232,6 +239,25 @@ final class ArrayBufferView[A](val array: Array[AnyRef], val length: Int) extend
 /** An object used internally by collections backed by an extensible Array[AnyRef] */
 object RefArrayUtils {
 
+  // We chose 8 as a good size for the minimum array
+  private[this] val minimalLength: Long = 8L
+
+  def reduceSize(array: Array[AnyRef], n: Int): Array[AnyRef] = {
```

Comment:
Is this method necessary? There is `java.util.Arrays.copyOf` for efficient copying to a different size.

Reply:
I suppose the case here is the same as for `ensureSize`. While `copyOf` may be more efficient for a full copy, we usually only want to copy a part of the array and leave the rest as `null`. I'll leave it at that for now.